File: crates/breez-sdk/cli/src/command/contacts.rs
use breez_sdk_spark::{AddContactRequest, BreezSdk, ListContactsRequest, UpdateContactRequest};
use clap::Subcommand;

use crate::command::print_value;

#[derive(Clone, Debug, Subcommand)]
pub enum ContactCommand {
    /// Add a new contact
    Add {
        /// Name of the contact
        name: String,
        /// Lightning address (user@domain)
        payment_identifier: String,
    },
    /// Update an existing contact
    Update {
        /// ID of the contact to update
        id: String,
        /// New name for the contact
        name: String,
        /// New Lightning address (user@domain)
        payment_identifier: String,
    },
    /// Delete a contact
    Delete {
        /// ID of the contact to delete
        id: String,
    },
    /// List contacts
    List {
        /// Number of contacts to skip
        offset: Option<u32>,
        /// Maximum number of contacts to return
        limit: Option<u32>,
    },
}

pub async fn handle_command(
    sdk: &BreezSdk,
    command: ContactCommand,
) -> Result<bool, anyhow::Error> {
    match command {
        ContactCommand::Add {
            name,
            payment_identifier,
        } => {
            let contact = sdk
                .add_contact(AddContactRequest {
                    name,
                    payment_identifier,
                })
                .await?;
            print_value(&contact)?;
            Ok(true)
        }
        ContactCommand::Update {
            id,
            name,
            payment_identifier,
        } => {
            let contact = sdk
                .update_contact(UpdateContactRequest {
                    id,
                    name,
                    payment_identifier,
                })
                .await?;
            print_value(&contact)?;
            Ok(true)
        }
        ContactCommand::Delete { id } => {
            sdk.delete_contact(id).await?;
            println!("Contact deleted successfully");
            Ok(true)
        }
        ContactCommand::List { offset, limit } => {
            let contacts = sdk
                .list_contacts(ListContactsRequest { offset, limit })
                .await?;
            print_value(&contacts)?;
            Ok(true)
        }
    }
}

File: crates/breez-sdk/bindings/examples/cli/langs/wasm/src/contacts.js
'use strict'

const { Command } = require('commander')
const { printValue } = require('./serialization')

/**
 * Register all contacts subcommands on the given commander program.
 *
 * @param {Command} program - The parent commander program (or subcommand)
 * @param {() => object} getSdk - Function that returns the SDK instance
 */
function registerContactsCommands(program, getSdk) {
  const contacts = program
    .command('contacts')
    .description('Contacts related commands')

  // --- add ---
  contacts
    .command('add')
    .description('Add a new contact')
    .argument('<name>', 'Name of the contact')
    .argument('<payment_identifier>', 'Lightning address (user@domain)')
    .action(async (name, paymentIdentifier) => {
      const sdk = getSdk()
      const contact = await sdk.addContact({
        name,
        paymentIdentifier
      })
      printValue(contact)
    })

  // --- update ---
  contacts
    .command('update')
    .description('Update an existing contact')
    .argument('<id>', 'ID of the contact to update')
    .argument('<name>', 'New name for the contact')
    .argument('<payment_identifier>', 'New Lightning address (user@domain)')
    .action(async (id, name, paymentIdentifier) => {
      const sdk = getSdk()
      const contact = await sdk.updateContact({
        id,
        name,
        paymentIdentifier
      })
      printValue(contact)
    })

  // --- delete ---
  contacts
    .command('delete')
    .description('Delete a contact')
    .argument('<id>', 'ID of the contact to delete')
    .action(async (id) => {
      const sdk = getSdk()
      await sdk.deleteContact(id)
      console.log('Contact deleted successfully')
    })

  // --- list ---
  contacts
    .command('list')
    .description('List contacts')
    .argument('[offset]', 'Number of contacts to skip')
    .argument('[limit]', 'Maximum number of contacts to return')
    .action(async (offset, limit) => {
      const sdk = getSdk()
      const contacts = await sdk.listContacts({
        offset: offset != null ? parseInt(offset, 10) : undefined,
        limit: limit != null ? parseInt(limit, 10) : undefined
      })
      printValue(contacts)
    })
}

module.exports = { registerContactsCommands }
